packages/server/lib/server-base.ts

```diff
@@ -520,6 +521,9 @@ export abstract class ServerBase<TSocket extends SocketE2E | SocketCt> {
           port,
           protocol,
         },
+        headers: {
+          'x-cypress-forwarded-from-proxy': true,
```

Comment:
This is a temporary workaround to fix a problem where the child cypress instance is not getting a connect event and thus, the socket is not added to the allow list. There will be a follow up PR with a proper fix for this.